### PR TITLE
[Reader Improvements] Implement new header for site

### LINKED ISSUE #19069: [Reader Improvements] Implement new design for site header
Implement new header design for selected site on Reader.

Design specs: pcdRpT-3BI-p2

### PR DESCRIPTION
**Pull request by RenanLukas**
### ~Do not merge: waiting for discussions: pcdRpT-3BI-p2#comment-5952 and p1693007089821289/1692821259.309009-slack-C05N140C8H5~

Fixes #19069 

To test:
1 - Open `"Me"` -> Debug settings -> `"Features in development"`;
2 - Enable `ReaderImprovementsFeatureConfig`;
3 - Move back to home and open Reader;
4 - Tap on any site's name/domain to open site posts list;
5 - You should see the new header layout. Make sure follow/unfollow button is working as expected;
<p float="left">
Screen Shot 2023-08-31 at 3 52 02 PM
Screen Shot 2023-08-31 at 3 51 56 PM
</p>

6 - Select a site without icon: the icon should be gone instead of displaying the placeholder;
7 - Repeat step 1 but this time disable `ReaderImprovementsFeatureConfig`;
8 - Repeat steps 3 and 4: you should see the existing header layout. Make sure follow/unfollow button is working as expected.
<p float="left">
Screen Shot 2023-08-25 at 11 11 12 PM
Screen Shot 2023-08-25 at 10 57 09 PM
</p>

## Regression Notes
1. Potential unintended areas of impact
None (implementation behind feature flag)

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing

3. What automated tests I added (or what prevented me from doing so)
None (all existing logic is inside a custom view)

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist:

- [x] Portrait and landscape orientations.
- [x] Light and dark modes.
- [x] Fonts: Larger, smaller and bold text.
- [x] High contrast.
- [ ] Talkback.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] Large and small screen sizes. (Tablet and smaller phones)
- [ ] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

### CODE REVIEW

**Comment by osullivanchris** [comment]
Looks good already! Some thoughts:

- button (style, color) and navigation bar (icons) are not per design yet, assuming separate changes?
- the navigation bar has very little inside it, so let's hide it on scroll, WDYT?
- the border around the image is needed if the image contains white/black. But takes away a bit from images that don't need it. Could we make it thinner, like 0.5pt? Is it a solid color or using some opacity?
- Can you check the text color values are all high emphasis (#000000 1.0) or medium emphasis (#000000 0.6). Some look slightly off to me, but I can't be sure by color picking a screenshot.

**Comment by RenanLukas** [comment]
Thanks for the review, @osullivanchris !

> button (style, color) and navigation bar (icons) are not per design yet, assuming separate changes?

Good catch! The follow/unfollow button will be handled in a [separate issue](https://github.com/wordpress-mobile/WordPress-Android/issues/19078), but I missed the navigation bar icon change (from "x" to arrow). I'll take care of it in this PR. Just to confirm: we're also removing the share button from the toolbar, correct? Like, before:
image

And after:
image


>the navigation bar has very little inside it, so let's hide it on scroll, WDYT?

Makes sense, done 👍 

> the border around the image is needed if the image contains white/black. But takes away a bit from images that don't need it. Could we make it thinner, like 0.5pt? Is it a solid color or using some opacity?

The value was 2dp (same as the existing design for this circle). If you agree I've changed it to `1dp` instead of `.5dp` because using decimals for density-independent pixels is usually not recommended. 
The color of the "border" is `#1A000000`, which should be black with 10% opacity. It's the color of the placeholder, and since a padding is being set to the image, the "border" color is actually from the placeholder behind it. Should I change it?

BTW, this is the placeholder:
image


> Can you check the text color values are all high emphasis (#000000 1.0) or medium emphasis (#000000 0.6). Some look slightly off to me, but I can't be sure by color picking a screenshot.

Should the "description" (in the screenshot above, the text `"The latest news on WordPress.com..."`) also be high emphasis? This is the only one that had medium emphasis on code. I'm pushing this change (description from medium emphasis to high emphasis) so you can have a look but there might be something off with my implementation. Could you please tell me which one(s) did you think were off?

**Comment by RenanLukas** [comment]
@osullivanchris in case it helps, I've updated the screenshots in the PR description with my latest changes.

**Review by osullivanchris — APPROVED** [review]
> Good catch! The follow/unfollow button will be handled in a https://github.com/wordpress-mobile/WordPress-Android/issues/19078, but I missed the navigation bar icon change (from "x" to arrow). I'll take care of it in this PR. Just to confirm: we're also removing the share button from the toolbar, correct? Like, before:

Yeah I intended to remove it. I don't think there's any reason for it to be here and so prominent. All it seems to do is share a link to that site. Not sure what the user task is here. I also think back icon is better. But let's make sure its using the right navigation pattern for back, rather than x presenting modally - if there is a difference.

> The value was 2dp (same as the existing design for this circle). If you agree I've changed it to 1dp instead of .5dp because using decimals for density-independent pixels is usually not recommended.
The color of the "border" is #1A000000, which should be black with 10% opacity. It's the color of the placeholder, and since a padding is being set to the image, the "border" color is actually from the placeholder behind it. Should I change it?

That sounds great. I'm happy with that. Yeah I'm aware 0.5 isn't recommended. The only exception I use is for hairline separators, but usually on iOS. Given it is 2pt in the current build, 1pt should be fine!

> Should the "description" (in the screenshot above, the text "The latest news on WordPress.com...") also be high emphasis? This is the only one that had medium emphasis on code. I'm pushing this change (description from medium emphasis to high emphasis) so you can have a look but there might be something off with my implementation. Could you please tell me which one(s) did you think were off?

That's right the description should be medium emphasis. It was actually the others, some of the high-emphasis/black colors looked slightly off from black. But it might just be because I'm color picking a pixelated png and picking it up wrong. Going to download the APK again now but looks good from the screenshots

**Comment by osullivanchris** [comment]
Build looks good too! That empty site icon is a bit random, using a compass logo I guess to mean web. 

WDYT about these options:
- hide icon if no icon (only text heading)
- show a letter in there, e.g. first letter of the site name
- update to a different icon e.g. below

Screenshot 2023-08-30 at 13 12 38

**Comment by RenanLukas** [comment]
@osullivanchris thanks for testing it again. 

>Build looks good too! That empty site icon is a bit random, using a compass logo I guess to mean web.

I agree 👍 

> WDYT about these options:
>
> - hide icon if no icon (only text heading)
> - show a letter in there, e.g. first letter of the site name
> - update to a different icon e.g. below

Options 1 and 3 should be very easy. Which one do you prefer?

**Comment by osullivanchris** [comment]
I think option 1 no icon might make most sense. Just depends if we can identify that there is no icon. For example, the placeholder set when site is created is not recognised as an icon on our side, if you know what I mean. That we know its a placeholder/no icon state.

**Comment by RenanLukas** [comment]
> I think option 1 no icon might make most sense. Just depends if we can identify that there is no icon. For example, the placeholder set when site is created is not recognised as an icon on our side, if you know what I mean. That we know its a placeholder/no icon state.

@osullivanchris I have a site that I never set an icon before, and I'm getting empty image URL in the Reader. Currently if the image URL is empty we just show the placeholder, so we can identify that there is no image URL and then hide the icon.

There's one thing, though: should we not only hide the icon in case there's no image set but also change the placeholder to something more meaningful? I suppose we still need a placeholder for users with bad connection, so we can show something in the circle while the image is being loaded, correct? If so, should I hide the icon but also change the placeholder to something like option 3? 

(edit)
I've pushed the option 1 changes (hiding the site icon if the URL is empty) and also the number of followers format agreed on pcdRpT-3BI-p2#comment-5978, so please feel free to check it out when you can. If you also want us to change the placeholder considering the bad internet connection scenario, I can do it 👍

**Comment by osullivanchris** [comment]
Having trouble accessing the build right now (might need to update my 2fa) but everything sounds good from the description 👍 

Agree we can update the icon for other cases. Here is a site-logo svg icon I found in the new library, which we can hopefully later on have as a central resource but for now just adding one by one.

site-logo

File: WordPress/src/main/res/values/dimens.xml [hunk 1]
```diff
@@ -238,6 +238,7 @@
     <dimen name="reader_chip_radius">8dp</dimen>
     <dimen name="reader_chip_height">44dp</dimen>
     <dimen name="reader_site_header_avatar_padding">2dp</dimen>
+    <dimen name="reader_site_header_avatar_padding_new">1dp</dimen>
```
Comment (develric) [inline WordPress/src/main/res/values/dimens.xml:241]: not blocking this PR and a sort of more generic thing to think about: maybe we should consider having a set of generic/meta padding dimensions, similar to what we have for margins [here](https://github.com/wordpress-mobile/WordPress-Android/blob/1b21010e8d3388fb963ef52263f8a23629a62430/WordPress/src/main/res/values/dimens.xml#L104).

**Review by thomashorta — APPROVED** [review]
The PR looks good to me. 

As discussed in p1694772256123939/1694771397.579109-slack-C01CW1VMLAF, I just re-added the Share button, as we are keeping it for sites.

**Review by develric — APPROVED** [review]
Thanks @RenanLukas for the PR and @thomashorta for taking it over (while Renan is AFK) to restore the share action as per agreement with design 🙇 . The PR LGTM and I'm merging it 👍 . 

I left an inline comment + 2 not blocking notes below for us to think about; but as said they are more on a general note side 🙇  :
- we should probably review in which context/screens it makes sense to align to the new icon [here](https://github.com/wordpress-mobile/WordPress-Android/pull/19081#issuecomment-1699050887). For example I see the site icon in the Manage Topics & Sites
image

- This is also a general thing to think about for similar situations where the base layout has elements potentially disappearing; using Charles to simulate slow connections with throttling while the site fields are fetched from the network, I sort of miss a smoother transition to compose the view and make the icon disappear, maybe some sort of loading state or shimmer skeleton (the scenario I forced is probably pretty edgy but might be something we can think of improve as a general pattern). cc @osullivanchris

https://github.com/wordpress-mobile/WordPress-Android/assets/47797566/1380cb47-9822-4346-be66-4e351d21c792

**Comment by osullivanchris** [comment]
> This is also a general thing to think about for similar situations where the base layout has elements potentially disappearing; using Charles to simulate slow connections with throttling while the site fields are fetched from the network, I sort of miss a smoother transition to compose the view and make the icon disappear, maybe some sort of loading state or shimmer skeleton (the scenario I forced is probably pretty edgy but might be something we can think of improve as a general pattern). cc @osullivanchris


That loading state isn't great in general. A skeleton shimmer design would be nice if feasible.

**Comment by RenanLukas** [comment]
Created issue to track shimmer effect implementation https://github.com/wordpress-mobile/WordPress-Android/issues/19235

**Pull request merged**